src/OrchardCore.Modules/OrchardCore.Https/Permissions.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using OrchardCore.Security.Permissions;
+
+namespace OrchardCore.Https
+{
+    public class Permissions : IPermissionProvider
+    {
+        public static readonly Permission ManageHttps = new Permission("ManageHttps", "Manage HTTPS");
+
+        public IEnumerable<Permission> GetPermissions()
+        {
+            return new[] { ManageHttps };
```

Comment:
Use `yield return ManageHttps` instead of creating new array each time

Reply:
I see. I don't see almost any usage of yield  through our codebase. We have a lot of "Permissions.cs" files and in all of them we are returning a new array.

I'm not sure of the performance tradeoff  of using yield for this. I would prefer filing an issue about that an let people decide about applying the change globally? Just for the sake of consistency.

Reply:
That's an interesting point. But in this case GetPermissions is called only once in the app per tenant (AFAIK, please correct me otherwise). Then you could say why not use a static field, but then we would still have "one" instance, and it would be worse because it would never get collected. It's only better, maybe, if we have lots of tenants.

Reply:
You can still file an issue to investigate what is the best pattern. But I don't think we lose much in terms of perf here.

Reply:
This is type inference on a static readonly field. There should be no perf issue. I guess it will just make the compiler work a little more but the outcome should be the same.